### DISCUSSION TITLE
Benefits of GitHub based Q&A system over mailing list

### DISCUSSION THREAD

**Discussion by pattacini** [robotology/community#118] [Feedback & Meta]
Just to give a look in retrospect, here are the main advantages we managed to gain by using GitHub as our Q&A system, as taken from https://www.godaddy.com/garage/webpro/development/cool-github-hack.
#### It's asynchronous

> For questions that fall into the “important, but not urgent” box, this model works really well. An individual can post a public question to another, and the recipient can get to it in the queue when time permits.
#### It enables arbitrary length of response

> Although some questions can be answered in 140 characters, others require a lengthier and more thought-full (as opposed to thought-free) response. This works wonderfully for that.
#### It’s persistent, one-to-many and public

> Both of the points above could be applied to email as well. However, engaging in Q&A using this mechanism enables the interaction to be **indexed by search engines** and enables it to be found by others in the future who might have a similar question.

No need for subscribing to read the threads: **everything is public** and, as such, problems and solutions can be easily ranked by Google.
#### It builds reputation and community

> All of the Qs and As that gather engagement not only show up in the Q&A thread, but also show up in the **activity history of the individual participants**. This gives a pulse to their activity over time when they participate. GitHub also highlights the group of individuals who have participated in the topic thread, giving the conversation a human face.

**Comment by alecive** [comment]
It doesn't clutter everyone's inbox :smile:

**Comment by lornat75** [comment]
Another positive point is that now that we have routed to github most of the specific questions from users, we can use the mailing list for important, general purpose notifications.

**Comment by drdanz** [comment]
We can easily **redirect questions** to the repositories they belong to, or assign them to someone.

**Comment by pattacini** [comment]
@drdanz actually, **assigning tasks** is a big pro.

**Comment by drdanz** [comment]
We can also fix typos, formatting, etc
